[BUG] undefined reference to `shm_open'. `-lrt` missed ?

**Describe the bug**
<!--A clear and concise description of what the bug is.-->
`make release` reports error.

**To Reproduce**
<!--Steps to reproduce the behavior:

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
-->
`cd bpftime && make release`

**Expected behavior**
<!--A clear and concise description of what you expected to happen.-->

**Screenshots**
<!--If applicable, add screenshots to help explain your problem.-->

```
gmake[3]: Entering directory '/home/pdliyan/bpftime/build'
[ 95%] Linking CXX executable bpftime_daemon
../runtime/libruntime.a(bpftime_shm_internal.cpp.o): In function `bpftime_remove_global_shm':
bpftime_shm_internal.cpp:(.text+0x4223): undefined reference to `shm_unlink'
../runtime/libruntime.a(bpftime_shm_internal.cpp.o): In function `bpftime::bpftime_shm::bpftime_shm(char const*, bpftime::shm_open_type)':
bpftime_shm_internal.cpp:(.text+0xc9dd): undefined reference to `shm_unlink'
../runtime/libruntime.a(bpftime_shm_internal.cpp.o): In function `boost::interprocess::shared_memory_object::shared_memory_object(boost::interprocess::open_only_t, char const*, boost::interprocess::mode_t)':
bpftime_shm_internal.cpp:(.text._ZN5boost12interprocess20shared_memory_objectC2ENS0_11open_only_tEPKcNS0_6mode_tE[_ZN5boost12interprocess20shared_memory_objectC5ENS0_11open_only_tEPKcNS0_6mode_tE]+0xa1): undefined reference to `shm_open'
../runtime/libruntime.a(bpftime_shm_internal.cpp.o): In function `void boost::interprocess::ipcdetail::managed_open_or_create_impl<boost::interprocess::shared_memory_object, 16ul, true, false>::priv_open_or_create<char const*, boost::interprocess::ipcdetail::create_open_func<boost::interprocess::ipcdetail::basic_managed_memory_impl<char, boost::interprocess::rbtree_best_fit<boost::interprocess::mutex_family, boost::interprocess::offset_ptr<void, long, unsigned long, 0ul>, 0ul>, boost::interprocess::iset_index, 16ul> > >(boost::interprocess::ipcdetail::create_enum_t, char const* const&, unsigned long, boost::interprocess::mode_t, void const*, boost::interprocess::permissions const&, boost::interprocess::ipcdetail::create_open_func<boost::interprocess::ipcdetail::basic_managed_memory_impl<char, boost::interprocess::rbtree_best_fit<boost::interprocess::mutex_family, boost::interprocess::offset_ptr<void, long, unsigned long, 0ul>, 0ul>, boost::interprocess::iset_index, 16ul> >)':
bpftime_shm_internal.cpp:(.text._ZN5boost12interprocess9ipcdetail27managed_open_or_create_implINS0_20shared_memory_objectELm16ELb1ELb0EE19priv_open_or_createIPKcNS1_16create_open_funcINS1_25basic_managed_memory_implIcNS0_15rbtree_best_fitINS0_12mutex_familyENS0_10offset_ptrIvlmLm0EEELm0EEENS0_10iset_indexELm16EEEEEEEvNS1_13create_enum_tERKT_mNS0_6mode_tEPKvRKNS0_11permissionsET0_[_ZN5boost12interprocess9ipcdetail27managed_open_or_create_implINS0_20shared_memory_objectELm16ELb1ELb0EE19priv_open_or_createIPKcNS1_16create_open_funcINS1_25basic_managed_memory_implIcNS0_15rbtree_best_fitINS0_12mutex_familyENS0_10offset_ptrIvlmLm0EEELm0EEENS0_10iset_indexELm16EEEEEEEvNS1_13create_enum_tERKT_mNS0_6mode_tEPKvRKNS0_11permissionsET0_]+0x138): undefined reference to `shm_open'
bpftime_shm_internal.cpp:(.text._ZN5boost12interprocess9ipcdetail27managed_open_or_create_implINS0_20shared_memory_objectELm16ELb1ELb0EE19priv_open_or_createIPKcNS1_16create_open_funcINS1_25basic_managed_memory_implIcNS0_15rbtree_best_fitINS0_12mutex_familyENS0_10offset_ptrIvlmLm0EEELm0EEENS0_10iset_indexELm16EEEEEEEvNS1_13create_enum_tERKT_mNS0_6mode_tEPKvRKNS0_11permissionsET0_[_ZN5boost12interprocess9ipcdetail27managed_open_or_create_implINS0_20shared_memory_objectELm16ELb1ELb0EE19priv_open_or_createIPKcNS1_16create_open_funcINS1_25basic_managed_memory_implIcNS0_15rbtree_best_fitINS0_12mutex_familyENS0_10offset_ptrIvlmLm0EEELm0EEENS0_10iset_indexELm16EEEEEEEvNS1_13create_enum_tERKT_mNS0_6mode_tEPKvRKNS0_11permissionsET0_]+0x295): undefined reference to `shm_open'
bpftime_shm_internal.cpp:(.text._ZN5boost12interprocess9ipcdetail27managed_open_or_create_implINS0_20shared_memory_objectELm16ELb1ELb0EE19priv_open_or_createIPKcNS1_16create_open_funcINS1_25basic_managed_memory_implIcNS0_15rbtree_best_fitINS0_12mutex_familyENS0_10offset_ptrIvlmLm0EEELm0EEENS0_10iset_indexELm16EEEEEEEvNS1_13create_enum_tERKT_mNS0_6mode_tEPKvRKNS0_11permissionsET0_[_ZN5boost12interprocess9ipcdetail27managed_open_or_create_implINS0_20shared_memory_objectELm16ELb1ELb0EE19priv_open_or_createIPKcNS1_16create_open_funcINS1_25basic_managed_memory_implIcNS0_15rbtree_best_fitINS0_12mutex_familyENS0_10offset_ptrIvlmLm0EEELm0EEENS0_10iset_indexELm16EEEEEEEvNS1_13create_enum_tERKT_mNS0_6mode_tEPKvRKNS0_11permissionsET0_]+0x448): undefined reference to `shm_open'
collect2: error: ld returned 1 exit status
gmake[3]: *** [daemon/CMakeFiles/bpftime_daemon.dir/build.make:103: daemon/bpftime_daemon] Error 1
gmake[3]: Leaving directory '/home/pdliyan/bpftime/build'
gmake[2]: *** [CMakeFiles/Makefile2:813: daemon/CMakeFiles/bpftime_daemon.dir/all] Error 2
gmake[2]: Leaving directory '/home/pdliyan/bpftime/build'
gmake[1]: *** [Makefile:136: all] Error 2
gmake[1]: Leaving directory '/home/pdliyan/bpftime/build'
make: *** [Makefile:54: release] Error 2
```

**Desktop (please complete the following information):**
* OS: Centos
* Version 8
CentOS Linux release 8.4.2105

**Additional context**
<!--Add any other context about the problem here.-->
Not familiar with `gmake`, have to send this bug report.
With `-lrt` flag, my local test code with `shm_open` works okay.
```
#include <stdio.h>
#include <fcntl.h>
#include <sys/mman.h>
#include <unistd.h>

int main() {
    // 创建或打开共享内存对象
    int fd = shm_open("/my_shared_memory", O_CREAT | O_RDWR, 0666);
    if (fd == -1) {
        perror("shm_open");
        return 1;
    }

    // 设置共享内存对象的大小
    if (ftruncate(fd, 1024) == -1) {
        perror("ftruncate");
        return 1;
    }

    // 将共享内存映射到进程的地址空间
    void* ptr = mmap(NULL, 1024, PROT_READ | PROT_WRITE, MAP_SHARED, fd, 0);
    if (ptr == MAP_FAILED) {
        perror("mmap");
        return 1;
    }

    // 在共享内存中写入数据
    sprintf((char*)ptr, "Hello, shared memory!");

    // 解除内存映射
    if (munmap(ptr, 1024) == -1) {
        perror("munmap");
        return 1;
    }

    // 关闭共享内存对象
    if (close(fd) == -1) {
        perror("close");
        return 1;
    }

    // 删除共享内存对象
    if (shm_unlink("/my_shared_memory") == -1) {
        perror("shm_unlink");
        return 1;
    }

    printf("okay\n");

    return 0;
}
```

```
$ gcc -o sha_e shm.c -lrt && ./sha_e
okay
```

--- a/daemon/CMakeLists.txt
+++ b/daemon/CMakeLists.txt
@@ -6,8 +6,8 @@ add_bpf_skel_generating_target(bpftime_daemon_ebpf_skel ${CMAKE_CURRENT_BINARY_D
 
 add_dependencies(bpftime_daemon_ebpf_skel bpftime_daemon_ebpf_target)
 
-add_library(libbpftime_daemon STATIC     
-    user/bpf_tracer.cpp 
+add_library(libbpftime_daemon STATIC
+    user/bpf_tracer.cpp
     user/handle_bpf_event.cpp
     user/bpftime_driver.cpp
 )
@@ -16,27 +16,29 @@ add_executable(bpftime_daemon
     user/main.cpp
 )
 
-add_dependencies(libbpftime_daemon 
-        bpftime_daemon_ebpf_skel 
-        libbpf 
-        spdlog::spdlog
-        runtime
-    )
+add_dependencies(libbpftime_daemon
+    bpftime_daemon_ebpf_skel
+    libbpf
+    spdlog::spdlog
+    runtime
+)
 
-target_include_directories(libbpftime_daemon PRIVATE 
-    ${CMAKE_CURRENT_BINARY_DIR} 
+target_include_directories(libbpftime_daemon PRIVATE
+    ${CMAKE_CURRENT_BINARY_DIR}
     ${LIBBPF_INCLUDE_DIRS}
     ${CMAKE_CURRENT_SOURCE_DIR}
     ${CMAKE_CURRENT_SOURCE_DIR}/../vm/include
     ${CMAKE_CURRENT_SOURCE_DIR}/../runtime/include
 )
-target_link_libraries(libbpftime_daemon PRIVATE 
-        ${LIBBPF_LIBRARIES}
-        elf
-        z
-        spdlog::spdlog
-        runtime
-    )
+target_link_libraries(libbpftime_daemon PRIVATE
+    ${LIBBPF_LIBRARIES}
+    elf
+    z
+    spdlog::spdlog
+    runtime
+    PUBLIC
+    rt
+)
 set_property(TARGET libbpftime_daemon PROPERTY CXX_STANDARD 20)
 
 add_dependencies(bpftime_daemon libbpftime_daemon)
@@ -45,6 +47,5 @@ target_link_libraries(bpftime_daemon PRIVATE libbpftime_daemon)
 install(TARGETS bpftime_daemon CONFIGURATIONS Release Debug DESTINATION ~/.bpftime)
 
 if(BPFTIME_ENABLE_UNIT_TESTING)
-  add_subdirectory(test)
+    add_subdirectory(test)
 endif()
-

